modules/text2vec-transformers/clients/meta.go

```diff
@@ -16,12 +16,86 @@ import (
 	"encoding/json"
 	"io/ioutil"
 	"net/http"
+	"sync"
 
 	"github.com/pkg/errors"
 )
 
+type metaInfoFn func() (map[string]interface{}, error)
+
 func (v *vectorizer) MetaInfo() (map[string]interface{}, error) {
-	req, err := http.NewRequestWithContext(context.Background(), "GET", v.url("/meta"), nil)
+	metaInfoFns := v.allMetaInfoFns()
+
+	switch len(metaInfoFns) {
+	case 0:
+		return nil, errors.Errorf("no metaInfo callback configured")
+	case 1:
+		for _, singleMetaInfo := range metaInfoFns {
+			return singleMetaInfo()
+		}
+	}
+	return v.metaInfoAll(metaInfoFns)
+}
+
+func (v *vectorizer) allMetaInfoFns() map[string]metaInfoFn {
+	if v.originPassage != v.originQuery {
+		return map[string]metaInfoFn{
+			"passage": func() (map[string]interface{}, error) { return v.metaInfo(v.urlPassage) },
+			"query":   func() (map[string]interface{}, error) { return v.metaInfo(v.urlQuery) },
+		}
+	}
+	return map[string]metaInfoFn{
+		"common": func() (map[string]interface{}, error) { return v.metaInfo(v.urlPassage) },
+	}
+}
+
+func (v *vectorizer) metaInfoAll(metaInfoFns map[string]metaInfoFn) (map[string]interface{}, error) {
+	type nameMeta struct {
+		name string
+		meta map[string]interface{}
+	}
+	type nameErr struct {
+		name string
+		err  error
+	}
+
+	var wg sync.WaitGroup
+	chMeta := make(chan nameMeta, len(metaInfoFns))
+	chErr := make(chan nameErr, len(metaInfoFns))
```

Comment:
maybe using just one channel for the meta response would be enough?

Reply:
reimplemented. no channels are used now